### PR TITLE
[TEVA-1784] Confirmation email for job application submit

### PR DESCRIPTION
**Pull request by david-mears-dfe**
## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-1784

## Changes in this PR:

- Adds the email
- Some light refactoring of mailers
- Updates seeds (was a separate PR https://github.com/DFE-Digital/teaching-vacancies/pull/2997 so feel free to ignore the last commit in this PR - I forgot that it was branched off this one so when I merged it it just went into this PR.)

### A job application at a central office of a trust

Screenshot 2021-03-02 at 16 44 25

### A job application at multiple schools in a local authority

Screenshot 2021-03-02 at 16 44 20

### A job application at a single school in a trust

Screenshot 2021-03-02 at 16 44 09

### CODE REVIEW

**Comment by david-mears-dfe** [comment]
https://teaching-vacancies-review-pr-2994.london.cloudapps.digital/

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-2994.london.cloudapps.digital>

**Comment by david-mears-dfe** [comment]
It also actually sends, if you tell it to on the review app:
Screenshot 2021-03-02 at 18 30 53

File: db/seeds.rb [hunk 1]
```diff
@@ -43,6 +43,27 @@
                                school_type: "Academy",
                                gias_data: { 'ReligiousCharacter (name)': "None" })
 
+# A second school at the local authority to enable seeding a vacancy at multiple schools.
```
Comment (david-mears-dfe) [inline db/seeds.rb:46]: Ignore this file's changes - they were reviewed in a separate PR which branched off this one (forgot that when I merged it!)

File: config/initializers/notifications.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 NOTIFY_KEY = ENV["NOTIFY_KEY"]
+NOTIFY_APPLICATION_SUBMITTED_CONFIRMATION_TEMPLATE = "0a75c46b-923f-4ee0-807c-99fd289a881a".freeze
```
Comment (cpjmcquillan) [inline config/initializers/notifications.rb:2]: Maybe `_JOBSEEKER_` would be useful, or maybe it's too cumbersome?

File: spec/system/jobseekers_can_submit_a_job_application_spec.rb [hunk 3]
```diff
@@ -1,6 +1,8 @@
 require "rails_helper"
 
 RSpec.describe "Jobseekers can submit a job application" do
+  include ActiveJob::TestHelper
```
Comment (cpjmcquillan) [inline spec/system/jobseekers_can_submit_a_job_application_spec.rb:4]: Do we want to include this by default?

File: spec/system/jobseekers_can_submit_a_job_application_spec.rb [hunk 4]
```diff
@@ -13,18 +15,23 @@
...
       check "Confirm data usage"
-
-      expect { click_on I18n.t("buttons.submit_application") }
+      expect { perform_enqueued_jobs { click_on I18n.t("buttons.submit_application") } }
         .to change { JobApplication.first.status }.from("draft").to("submitted")
+        .and change { delivered_emails.count }.by(1)
+
       expect(page).to have_content(I18n.t("jobseekers.job_applications.submit.panel.title"))
+
+      # TODO: expect that the first link goes to the view your application path, once that page has been created
```
Comment (cpjmcquillan) [inline spec/system/jobseekers_can_submit_a_job_application_spec.rb:32]: This page has been created now

File: db/seeds.rb [hunk 5]
```diff
@@ -43,6 +43,27 @@
...
+                                 county: "Hampshire",
+                                 postcode: "SO18 2NX",
+                                 region: "London",
+                                 easting: "445283",
+                                 northing: "114779",
+                                 geolocation: "(50.9306936449461,-1.3569968052135)",
+                                 readable_phases: %w[primary],
+                                 detailed_school_type: "Foundation school",
+                                 school_type: "Local authority maintained schools",
+                                 gias_data: { 'ReligiousCharacter (name)': "Does not apply" })
```
Comment (cpjmcquillan) [inline db/seeds.rb:65]: What's the reason for single quotes here?

File: config/locales/mailers.yml [hunk 6]
```diff
@@ -33,6 +33,22 @@ en:
...
+      intro: Your application for %{job_title} at %{organisation_name}
+      more_info:
+        heading: Still have questions?
+        description: If you would like more information you can contact the school at %{email}.
+      next_steps:
+        heading: Next steps
+        description: >-
+          You can %{link} on your account. If you are shortlisted for an interview, %{organisation_name} will be in
+          touch to let you know.
+        link_text: view your application (TODO - update this link to point to the right place once that page has been created)
```
Comment (cpjmcquillan) [inline config/locales/mailers.yml:47]: This page has been created

**Review by cpjmcquillan — APPROVED** [review]

File: config/initializers/notifications.rb (hunk 2 above)
Comment (david-mears-dfe) [reply config/initializers/notifications.rb:2]: It's a fair cop.

File: spec/system/jobseekers_can_submit_a_job_application_spec.rb (hunk 3 above)
Comment (david-mears-dfe) [reply spec/system/jobseekers_can_submit_a_job_application_spec.rb:4]: Maybe so

File: spec/system/jobseekers_can_submit_a_job_application_spec.rb (hunk 4 above)
Comment (david-mears-dfe) [reply spec/system/jobseekers_can_submit_a_job_application_spec.rb:32]: But it's easier to write placeholder code than real code!

File: db/seeds.rb (hunk 5 above)
Comment (david-mears-dfe) [reply db/seeds.rb:65]: I blame Cesi

**Pull request merged**